src/python/pants/python/python_setup.py

```diff
@@ -109,6 +109,17 @@ def register_options(cls, register):
                 "and run `./pants lock ::`."
             ),
         )
+        register(
+            "--fail-on-invalid-lockfile",
```

Comment:
We've historically used an enum for this option, something like `--stale-lockfile-behavior={warn,error}`. That started because we often had `{ignore,warn,error}`. I doubt we want `ignore` to be an option, as that's not very safe to do..so it's less obvious an enum is necessary

But I think I'd still bias towards it for the sake of consistency and also the option being more self-documenting. No need to explain "The default behavior...", that will be obvious

Reply:
That makes sense.